src/pkgcheck/checks/reserved.py

```diff
@@ -0,0 +1,94 @@
+import re
+from .. import addons, bash, results, sources
+from . import Check
+
+
+class _ReservedNameCheck(Check):
+    reserved_prefixes = ('__', 'abort', 'dyn', 'prep')
+    reserved_substrings = ('hook', 'paludis', 'portage')  # 'ebuild' is special case
+    reserved_ebuild_regex = re.compile(r'(.*[^a-zA-Z])?ebuild.*')
+
+    """Portage variables whose use is half-legitimate and harmless if the package manager doesn't support them."""
+    special_whitelist = ('EBUILD_DEATH_HOOKS', 'EBUILD_SUCCESS_HOOKS', 'PORTAGE_QUIET')
+
+    def _check(self, used_type: str, used_names):
+        for used_name, node_start in used_names.items():
+            if used_name in self.special_whitelist:
+                continue
+            test_name = used_name.lower()
```

Comment:
I'm nitpicking a little bit but I think we should note that the current logic assumes exemptions will only be uppercase which is why it's laid out like this.

Reply:
I think the exemptions (whitelist) is a specific case by case one. Maybe `EBUILD_DEATH_HOOKS` is fine, but `ebuild_death_hooks` is not. This whitelist is somewhat a work in progress, increased when needed.

Reply:
Suppose we have the tests anyway. Alright!